client/my-sites/controller.js

```diff
@@ -169,7 +170,9 @@ module.exports = {
 		const onSelectedSiteAvailable = () => {
 			const selectedSite = sites.getSelectedSite();
 			siteStatsStickyTabActions.saveFilterAndSlug( false, selectedSite.slug );
-			context.store.dispatch( receiveSite( selectedSite ) );
+			if ( ! getSite( context.store.getState(), selectedSite.ID ) ) {
+				context.store.dispatch( receiveSite( selectedSite ) );
```

Comment:
>This is an odd conditional. Would we be getting a `selectedSite` if it didn't already exist in the state tree?

I added some clarification to my comment at https://github.com/Automattic/wp-calypso/pull/10477#issuecomment-271590520 . Essentially we want to receive the site object only if we don't already have it tracked in Redux state. Otherwise we risk receiving in stale data if changes have since been made to the Redux state site.